src/pykeen/datasets/remote_literal_base.py

```diff
@@ -0,0 +1,216 @@
+        :param numeric_triples_preprocessing: function for preprocessing numeric attributive triples, defaults to None
+               e.g. ..utils.filter_triples_by_relations() can be used or a custom function to add/remove/edit triples
+        :param numeric_triples_preprocessing_kwargs: args to pass to the above preprocessing function, defaults to None
+        :param numeric_literals_preprocessing: function for preprocessing numeric literals, defaults to None
+               e.g. ..utils.minmax_normalize() can be used or a custom function to modify literals
+        :param numeric_literals_preprocessing_kwargs: args to pass to the above preprocessing function, defaults to None
+        :param kwargs: Passed to the superclass
+        """
+        super().__init__(**kwargs)
+
+        self.numeric_triples_url = numeric_triples_url
+        self.numeric_triples_preprocessing = numeric_triples_preprocessing
+        self.numeric_triples_preprocessing_kwargs = numeric_triples_preprocessing_kwargs
+        self.numeric_literals_preprocessing = numeric_literals_preprocessing
+        self.numeric_literals_preprocessing_kwargs = numeric_literals_preprocessing_kwargs
+
+        self.numeric_triples_file_name = name_from_url(self.numeric_triples_url)
+        self.path_to_numeric_triples = self.cache_root.joinpath(self.numeric_triples_file_name)  # noqa
+
+    # docstr-coverage: inherited
+    def _load(self) -> None:  # noqa: D102
+        """Load train & test relation triples from remote zip file and numeric attr. triples from remote text file."""
+        all_unpacked = all(path.is_file() for path in self._get_paths())
+
+        if not all_unpacked:
+            archive_file = self._get_bytes()
+            self._extract(archive_file=archive_file)
+            logger.info(f"Extracted to {self.cache_root}.")
+
+        download(url=self.numeric_triples_url, path=self.path_to_numeric_triples)
+
+        self._training = self.triples_factory_cls.from_path(
+            path=self.training_path,
+            create_inverse_triples=self._create_inverse_triples,
+            path_to_numeric_triples=self.path_to_numeric_triples,
+            numeric_triples_preprocessing=self.numeric_triples_preprocessing,
+            numeric_triples_preprocessing_kwargs=self.numeric_triples_preprocessing_kwargs,
+            numeric_literals_preprocessing=self.numeric_literals_preprocessing,
+            numeric_literals_preprocessing_kwargs=self.numeric_literals_preprocessing_kwargs,
+        )
+        self._testing = self.triples_factory_cls.from_path(
+            path=self.testing_path,
+            entity_to_id=self._training.entity_to_id,  # share entity index with training
+            relation_to_id=self._training.relation_to_id,  # share relation index with training
+            create_inverse_triples=self._create_inverse_triples,
+            path_to_numeric_triples=self.path_to_numeric_triples,
```

Comment:
the numeric triples are the same for training and testing?

Reply:
Yes, also in the original LiteralE implementation [all of them are loaded and used as one object](https://github.com/SmartDataAnalytics/LiteralE/blob/0b0c48fd9b74bf000400199610275ea5c159a44c/main_literal.py#L115).